scripts/deploy-pattern.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -o pipefail
+
+RUNS=5
+# Retry five times because the CRD might not be fully installed yet
+echo -n "Installing pattern: "
+for i in $(seq 1 ${RUNS}); do \
+    exec 3>&1 4>&2
+    OUT=$( { helm template --include-crds --name-template $* 2>&4 | oc apply -f- 2>&4 1>&3; } 4>&1 3>&1)
+    exec 3>&- 4>&-
+    ret=$?
+    if [ ${ret} -eq 0 ]; then
+        break;
+    else
+        echo -n "."
+        sleep 10
+    fi
+done
+
+# All the runs failed
+if [ ${i} -eq ${RUNS} ]; then
```

Comment:
Should we also check for an error condition here?

Reply:
Which scenario did you have in mind? I mean if helm template failed in a run but not in the last one, then we're good. If it failed also in the last run we show the error of the last run. So the scenario that we do not cover is where you have error A in previous run and error B in the last run, because we will show only error B. That should not be too horrible? (Unless I am missing something that is :) )

Reply:
I'm not well-versed in the semantics of pipefail, so I was thinking it might cause an early bailout. In general I think we're only really interested in the last error.

Reply:
Yeah normally bash only returns the exit code of the last command in a pipe, with pipefail it errors out even when the other ones fail. E.g. "false | true" returns 0 without pipefail and 1 with pipefail.